tools/autograd/gen_variable_type.py

```diff
@@ -734,6 +829,11 @@ def requires_unpack(arg):
                 suffix=suffix,
                 ref='&' if ref else '',
             ))
+
+            if dynamic_type == 'TensorList':
```

Comment:
why do we need to store this instead of just looking at unpacked args and seeing if they are Tensors or TensorLists?